iroh/src/commands/start.rs

```diff
@@ -200,14 +159,33 @@ pub(crate) async fn start_node(
         }
     }
 
-    let blob_dir = IrohPaths::BaoFlatStoreDir.with_root(iroh_data_root);
+    let blob_dir = IrohPaths::BaoStoreDir.with_root(iroh_data_root);
     let peers_data_path = IrohPaths::PeerData.with_root(iroh_data_root);
     tokio::fs::create_dir_all(&blob_dir).await?;
-    let root = iroh_data_root.to_path_buf();
-    tokio::task::spawn_blocking(|| migrate_flat_store_v0_v1(root)).await??;
-    let bao_store = iroh_bytes::store::flat::Store::load(&blob_dir)
+    let bao_store = iroh_bytes::store::redb::Store::load(&blob_dir)
         .await
         .with_context(|| format!("Failed to load iroh database from {}", blob_dir.display()))?;
+    let v0 = bao_store
+        .import_flat_store(FlatStorePaths {
+            complete: iroh_data_root.join("blobs.v0"),
+            partial: iroh_data_root.join("blobs-partial.v0"),
+            meta: iroh_data_root.join("blobs-meta.v0"),
+        })
+        .await?;
+    let v1 = bao_store
+        .import_flat_store(FlatStorePaths {
+            complete: iroh_data_root.join("blobs.v1").join("complete"),
+            partial: iroh_data_root.join("blobs.v1").join("partial"),
+            meta: iroh_data_root.join("blobs.v1").join("meta"),
+        })
+        .await?;
+    if v0 || v1 {
+        tracing::info!("flat data was imported - reapply inline options");
+        bao_store
+            .update_inline_options(InlineOptions::default(), true)
+            .await?;
+    }
```

Comment:
is the above the way I would implement migration in other places, like the node.rs in iroh-ffi?

Reply:
Yes, it's not really migration but import. You just tell it the location of a flat store and it will consume it and integrate it.

Once we are in redb land we would have internal migrations, but for the flat -> redb step it does not make sense because so many things change.

Reply:
makes sense, would be good to add that maybe to the PR description, so we have it in the changelog/a place to look up what the recommended structure is